catalog/CatalogDatabaseCache.cpp

```diff
@@ -61,15 +61,17 @@ void CatalogDatabaseCache::update(const serialization::CatalogDatabase &proto) {
 
   SpinSharedMutexExclusiveLock<false> lock(relations_mutex_);
   for (int i = 0; i < proto.relations_size(); ++i) {
-    auto relation_schema = make_unique<const CatalogRelationSchema>(proto.relations(i));
-    const relation_id rel_id = relation_schema->getID();
+    const serialization::CatalogRelationSchema &proto_relation = proto.relations(i);
+    const relation_id rel_id = proto_relation.relation_id();
```

Comment:
@hbdeshmukh This function is called once per query, before Foreman sends the first `WorkOrder` for the query.

You're right that we need to mutex protection once we have multiple query support (i.e., once `QueryManager` has merged), and your proposed refactor looks good. So, I prefer to do so in another PR. Thanks!